python/amici/pandas.py

```diff
@@ -367,28 +367,27 @@ def _get_names_or_ids(model, variable, by_id):
     # check whether variable type permitted
     variable_options = ['Parameter', 'FixedParameter', 'Observable', 'State']
     if variable not in variable_options:
-        raise ValueError('variable must be in ' + str(variable_options))
+        raise ValueError('Variable must be in ' + str(variable_options))
 
-    # functions to extract attributes
-    namegetter = getattr(model, 'get' + variable + 'Names')
-    idgetter = getattr(model, 'get' + variable + 'Ids')
+    # extract attributes
+    names = list(getattr(model, f'get{variable}Names')())
+    ids = list(getattr(model, f'get{variable}Ids')())
 
     # extract labels
-    if not by_id and len(set(namegetter())) == len(namegetter()) \
-            and getattr(model, f"has{variable}Names")():
+    if not by_id and len(set(names)) == len(names):
         # use variable names
-        return list(namegetter())
-    elif getattr(model, f"has{variable}Ids")():
+        return names
+    elif len(ids) > 0 or len(ids) == 0 and len(names) == 0:
```

Comment:
I believe this rather is a bug with has{Variable}Ids, I would rather change the implementations of has{Variable}Ids and has{Variable}Names unless @dweindl still remembers why they are implemented the way they currently are.

Reply:
more specifically change `&&` to `||` in the c++ code.